### DISCUSSION TITLE
posts/2022/05/webmentions-yes-javascript-no/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#13] [Announcements]
# posts/2022/05/webmentions-yes-javascript-no/

I wondered why a Hugo site needed JavaScript to go IndieWebbin’. I decided it didn’t.

https://www.brycewray.com/posts/2022/05/webmentions-yes-javascript-no/

**Comment by kaushalmodi** [comment]
> I present to you what could be the world’s only Hugo-based site which displays webmentions without requiring a single line of JavaScript.

Sorry.. I have been using getJSON based approach for few years on [my site](https://scripter.co) .. [reference](https://gitlab.com/kaushalmodi/hugo-theme-refined/-/blob/master/layouts/partials/webmention_rcv.html). You might find more examples on https://indieweb.org/Hugo too.

**Reply by brycewray** [reply]
I will make that correction! Thanks for letting me know.

**Edit**: Correction made.

**Comment by synesthesia** [comment]
I was going to write that I don't use javascript before pausing to remember that [the scripts I use](https://github.com/synesthesia/site2019/blob/master/scripts) are exactly that, however I run them at build time to store local data which I then pull into Hugo...

And to be even more honest, mine are built on a strong base from [Rowan Manning](https://github.com/rowanmanning/rowanmanning.com)!

**Reply by brycewray** [reply]
You make a good point. To clarify, my site does use [a little JS (for the purpose of inserting YouTube videos)](https://www.brycewray.com/posts/2021/09/gems-in-rough-09/#privacy-respecting-youtubeembeds), just not for webmentions as of yesterday.

Thanks for the mention of Rowan Manning’s site (Hugo-based), where he posted a [great article](https://rowanmanning.com/posts/webmentions-for-your-static-site/) about having webmentions.
